### PR TITLE
Documentation: Add link checker to CI and fix broken links

### PR DESCRIPTION
**Pull request by amotl**
Hi there,

this patch improves the state of the onion on all links within the documentation.

- Add [`crate-docs`](https://github.com/crate/crate-docs) build system.
- Run link checker earch morning.
- Rephrase a few spots in order to satisfy the Vale prose linter.

With kind regards,
Andreas.

### CODE REVIEW

**Review by mfussenegger — APPROVED** [review]

File: CHANGES.txt [hunk 1]
```diff
@@ -84,11 +84,13 @@ Unreleased
   If you are using this package with Python 2.7 or 3.3 already, you will not be
   able to install newer versions of this package.
 
+- Add support for SQLAlchemy 1.2
+
```
[the remark this answers is not in the document]
Comment (amotl) [reply CHANGES.txt:88]: I traced the addition where support for SQLAlchemy 1.2 was added on behalf of `git log` and then added this changelog item retroactively.

File: docs/appendices/compatibility.rst [hunk 2]
```diff
@@ -19,47 +19,11 @@ Version notes
...
 However, if you are running an older version of Python, older versions of the
 CrateDB Python client library can be used.
 
-.. CAUTION::
-
-    The documentation is written for Python 3.
-
-    If you are using older versions of this client library with Python 2, you
-    may find that you need to make adaptations to the instructions and sample
-    code to get things working.
```
[the remark this answers is not in the document]
Comment (amotl) [reply docs/appendices/compatibility.rst:33]: > Would be cool to not mix different aspects into one commit. Removing compat notes is fairly unrelated to adding a link checker and fixing broken links.

True. I will submit it on behalf of a different patch.

[the remark this answers is not in the document]
Comment (amotl) [reply docs/appendices/compatibility.rst]: See #410.

File: CHANGES.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (mfussenegger) [reply CHANGES.txt]: https://zignar.net/2020/11/14/why-you-should-keep-a-clean-git-history/  🙃

[the remark this answers is not in the document]
Comment (amotl) [reply CHANGES.txt]: Thank you for the heads up. I tried to separate concerns, see #410.

**Pull request merged**